Lib/test/test_cmd_line.py

```diff
@@ -89,12 +97,20 @@ def get_xoptions(*args):
     @unittest.skipIf(interpreter_requires_environment(),
                      'Cannot run -E tests when PYTHON env vars are required.')
     def test_unknown_xoptions(self):
-        rc, out, err = assert_python_failure('-X', 'blech')
+        _, out, err = assert_python_failure('-X', 'blech')
         self.assertIn(b'Unknown value for option -X', err)
-        msg = b'Fatal Python error: Unknown value for option -X'
+        msg = b'Fatal Python error: Unknown value for option -X (see -X help)'
```

Comment:
Lines 101 and 120 seem redundant but I kept both.

Reply:
It might be nice if this could print all the offending `-X` options, but that seems difficult given the interface to `_Py_check_xoptions()` and `config_read()`.

Reply:
Hmm I think something can be shown, either the unknown value or the raw argv.  I’ll have a look.

Reply:
Can’t show unknown value, _Py_check_xoptions returns a known string or NULL when invalid.

Reply:
Turns out the get xoptions method returns NULL for valid options, otherwise the invalid value!
I just need help to format the error string: I don’t know how to format the wchar_t string `option` into a const char error message.

```c
    /* -X options */
    const wchar_t* option = _Py_check_xoptions(&config->xoptions, known_xoptions);
    char errmsg[100];
    if (option != NULL) {
        // XXX gibberish printed with %S, %.50S, %ls + type conversion errors (same with snprintf)
        sprintf(errmsg, "Unknown value for option -X: %ls (see -X help)", option);
        return PyStatus_Error(errmsg);
    }
```

Reply:
> I just need help to format the error string: I don’t know how to format the wchar_t string option into a const char error message.

Wouldn't it be nicer if this could all be done in Python? 😄 

If it's difficult, don't worry about.

Reply:
It’s probably simple string conversion and formatting to a C programmer, but I am only playing one on TV here.

Reply:
@serhiy-storchaka my issue was with C string handling here!  thanks for the help

Reply:
`%ls` should work (and it is already used in other places. But if there are any issues with this, you can just omit outputting the option.

Reply:
I have seen that, and have tried it (see my message `Turns out etc`), but I couldn’t figure out the right combination of wchar_t, `L""` string and sprintf codes.

I will abandon printing the bad option, but anyway the window for 3.11 is missed.

Reply:
It is a nice enhancement and I hope to see it in 3.11. @pablogsal

Reply:
I would be happy to backport this. Please, add me as a reviewer in the 3.11 backport